packages/functions/src/optionSchemas.ts

```diff
@@ -168,6 +168,11 @@ export const optionSchemas: Record<string, CustomFunctionOptionsSchema> = {
         default: false,
         description: 'Returns all errors when true; otherwise only returns the first error.',
       },
+      uniqueId: {
+        type: 'object',
+        description:
+          'Assigns a unique id (by reference to the JS object) to a schema. It is used to optimize the creation of a validation function for a given schema; a given function will be stored by given id and retrieved between execution of validation.',
+      },
```

Comment:
Due to fact that `instances` in the `schema` function is a WeakMap's instance, we cannot use `string` type only `object` or `array`, which are references in JS. WDYT about it? I don't think so that we should change `instances` to the Map's instance (to support strings), because then we can end up with stackoverflow.

cc @P0lip

Reply:
Fair enough. Just make sure to add `x-internal` to indicate the option isn't meant to be used in rulesets.
```suggestion
      uniqueId: {
        type: 'object',
        description:
          'Assigns a unique id (by reference to the JS object) to a schema. It is used to optimize the creation of a validation function for a given schema; a given function will be stored by given id and retrieved between execution of validation.',
        'x-internal': true,
      },
```

Reply:
Thanks! Pushed.